### ISSUE TITLE
More flexible greenhouse model

### ISSUE THREAD

**Issue by lortordermur** [lortordermur/sfcalcsheet#1]
So far SFCalcSheet uses the equation from [idealized greenhouse model](https://en.wikipedia.org/wiki/Idealized_greenhouse_model) to calculate a planet’s surface temperature from its effective temperature.

Effective temperature:

![formula](https://latex.codecogs.com/gif.latex?T%20_%7Be%7D%3D%5Csqrt%5B4%5D%7B%5Cfrac%7B%281-A%29%5Ccdot%20L%7D%7B16%5Cpi%20%5Ccdot%20d%5E2%5Ccdot%20%5Csigma%20%7D%7D)

T(e) = Effective temperature (K), A = Bond albedo of planet, L = Luminosity of star (W), d = Distance to star (m), σ = Stefan-Boltzmann constant

Surface temperature (greenhouse model):

![formula](https://latex.codecogs.com/gif.latex?T%20_%7Bs%7D%3DT%20_%7Be%7D%5Ccdot%20%5Csqrt%5B4%5D%7B%5Cfrac%7B1%7D%7B1-%5Cfrac%7B%5Cepsilon%20%7D%7B2%7D%7D%7D)

T(s) = Surface temperature (K), T(e) = Effective temperature (K), ε = Atmospheric absorption (0-1)

The only atmospheric variable in this equation is *atmospheric absorption*/*emissivity*, a measure for how much radiation is retained via greenhouse gasses. Basically it only allows for modeling Earth’s atmosphere with varying amounts of greenhouse gasses. An equation that also takes air pressure would be useful for modeling worlds with a different atmosphere or a runaway greenhouse effect.

If you find something that could be adapted for use in SFCalcSheet, let me know here.

**Comment by lortordermur** [comment]
Here are some resources that may be helpful in modeling a H₂O and CO₂ based greenhouse effect on rocky planets, as well as possible ranges for a circumstellar habitable zone based on atmospheric properties.

Kasting et al.: [Habitable zones around main sequence stars](https://www.researchgate.net/publication/11809380_Habitable_zones_around_main_sequence_stars)
Ramses Mario Ramirez: [A more comprehensive habitable zone for finding life on other planets](https://www.researchgate.net/publication/326621666_A_more_comprehensive_habitable_zone_for_finding_life_on_other_planets)

**Comment by jchylik** [comment]
Perhaps a good question which greenhouse gasses do you consider. 
Although water vapour and carbon dioxide are the main greenhouse gasses in current Earth's atmosphere, there are also other greenhouse gasses that played an important role in the past (methane), and similar could be said for alternative scenario.

For example in a dysoptian sci-fi setting, you can consider a significant greenhouse contribution from nitrous oxides and CFCs (chlorofluorcarbons).

**Comment by jchylik** [comment]
And some relevant sources on values

Nitrous oxide and Ozone:
https://www.researchgate.net/publication/257864234_Emissivity_of_the_main_greenhouse_gases

CFCs:
https://doi.org/10.1016/0022-4073(77)90116-9

**Comment by lortordermur** [comment]
Thank you for your effort. I had a look at the sources soon after you posted them and they are really useful in respect to the relative contribution of various greenhouse gasses.

What is still missing for a possible update to the planetary habitability calculator is a simple-ish mathematical relationship between atmospheric density/pressure and overall intensity of the greenhouse effect (preferably as a temperature modificator). It is a very complex topic to research into but I will be able to set aside some time for it very soon and post any useful findings here.

**Comment by jchylik** [comment]
> a simple-ish mathematical relationship between atmospheric density/pressure and overall intensity of the greenhouse effect 

That is an interesting idea, but I have to disappoint you, such direct dependence is possible only for some gasses. However the concentration of some of the most important greenhouse gasses, such as water vapour, is very much altitude and temperature dependent.

Not the mention the effect of water vapour when it start condensing, ie. forming clouds.

**Comment by lortordermur** [comment]
@jchylik: Yeah, a proper greenhouse model would be way too complex for now. However you mentioned that you wanted to implement a calculation for radiation effects on the atmosphere? Could this be added to the "Planetary Habitability" calculator or would we better start a new one?

**Comment by jchylik** [comment]
> Could this be added to the "Planetary Habitability" calculator or would we better start a new one?

It could be added to the existing list, but it would take quite manu cells in order to calculate even a simplified radiative transfer model.

**Comment by lortordermur** [comment]
I think sheets can be hidden in LibreOffice (Sheet -> Hide Sheet), so the calculations could be performed in an additional hidden sheet named “model calculations” or similar.

**Comment by lortordermur** [comment]
It is possible to reference cells in another sheet: [Referencing a Cell in Another Sheet](https://help.libreoffice.org/latest/en-US/text/scalc/guide/cellreferences.html)

**Comment by jchylik** [comment]
> so the calculations could be performed in an additional hidden sheet named “model calculations”

Cool, I have used cross-sheet reference many times before, but hiding sheet is new to me.

**Comment by lortordermur** [comment]
Yes, I just discovered it by poking around in the GUI :-)

Do you have an idea what input values the model would take? Some of those would probably be already in the existing calculator.

**Comment by jchylik** [comment]
> Do you have an idea what input values the model would take? Some of those would probably be already in the existing calculator.

It would use things that got inserted in section "Physical properties", "Planetary habitability", as well as "Star! Star Color", and then we would just need to insert concentrations of some of the gasses in a couple of layers of atmosphere (with the initial values being those of earth), plus atmospheric pressure on the ground (in order to determine how much atmosphere does the planet have).

**Comment by lortordermur** [comment]
Due to the number of inputs that sounds like a new “Planetary Climate” calc would be in order.

The calculators are meant to be used manually in tandem, so one would transfer an output of one to an input of the model rather than reference another calculator directly. It might also be necessary to duplicate some code.

If you can give a list of input and output fields I could design a GUI and append the work copy here.

**Comment by jchylik** [comment]
OK, I will have a closer look at it in the upcoming days.

**Comment by lortordermur** [comment]
Great! We can flesh it out together then.

Next on the list would be star metallicity and after that the next release.

**Comment by lortordermur** [comment]
Did you have some time to look into it? If not yet, I would do an interim maintenance release.

**Comment by jchylik** [comment]
> Did you have some time to look into it?

Not yet, sorry, (I was moving to a new workplace).
Go ahead with a release, I will get to it hopefully in the next two weeks.

**Comment by lortordermur** [comment]
No problem, I will prepare the release then soon (I am currently working on a hobby game programming project quite intensely but I'll set aside a day).
